Code/Framework/AzToolsFramework/AzToolsFramework/Archive/ArchiveComponent.cpp

```diff
@@ -141,320 +120,474 @@ namespace AzToolsFramework
     }
 
-    void ArchiveComponent::CancelTasks(AZ::Uuid taskHandle)
-    {
-        AZStd::unique_lock<AZStd::mutex> lock(m_threadControlMutex);
 
-        auto it = m_threadInfoMap.find(taskHandle);
-        if (it == m_threadInfoMap.end())
+    std::future<bool> ArchiveComponent::AddFilesToArchive(
+        const AZStd::string& archivePath,
+        const AZStd::string& workingDirectory,
+        const AZStd::string& listFilePath)
+    {
+        if (!CheckParamsForAdd(workingDirectory, listFilePath))
         {
-            return;
+            std::promise<bool> p;
+            p.set_value(false);
+            return AZStd::move(p.get_future());
         }
 
-        ThreadInfo& info = it->second;
-        info.shouldStop = true;
-        m_cv.wait(lock, [&info]() {
-            return info.threads.size() == 0;
-        });
-        m_threadInfoMap.erase(it);
-    }
-    
-    void ArchiveComponent::LaunchZipExe(const AZStd::string& exePath, const AZStd::string& commandLineArgs, const ArchiveResponseOutputCallback& respCallback, AZ::Uuid taskHandle, const AZStd::string& workingDir, bool captureOutput)
-    {
-        auto sevenZJob = [=]()
+        auto FnAddFilesToArchive = [this, archivePath, workingDirectory, listFilePath](std::promise<bool>&& p) -> void
         {
-            if (!taskHandle.IsNull())
+            auto archive = m_archive->OpenArchive(archivePath);
+            if (!archive)
             {
-                AZStd::unique_lock<AZStd::mutex> lock(m_threadControlMutex);
-                m_threadInfoMap[taskHandle].threads.insert(AZStd::this_thread::get_id());
-                m_cv.notify_all();
+                AZ_Error(s_traceName, false, "Failed to open archive file '%s'", archivePath.c_str());
+                p.set_value(false);
+                return;
             }
 
-            AzFramework::ProcessLauncher::ProcessLaunchInfo info;
-            info.m_commandlineParameters = exePath + " " + commandLineArgs;
-            
-            info.m_showWindow = false;
-            if (!workingDir.empty())
-            {
-                info.m_workingDirectory = workingDir;
-            }
-            AZStd::unique_ptr<AzFramework::ProcessWatcher> watcher(AzFramework::ProcessWatcher::LaunchProcess(info, AzFramework::ProcessCommunicationType::COMMUNICATOR_TYPE_STDINOUT));
+            bool success = true; // starts true and turns false when any error is encountered.
+            AZ::IO::Path basePath{ workingDirectory };
 
-            AZStd::string consoleOutput;
-            AZ::u32 exitCode = static_cast<AZ::u32>(SevenZipExitCode::UserStoppedProcess);
-            if (watcher)
+            auto PerLineCallback = [&success, &basePath, &archive](AZStd::string_view filePathLine) -> void
             {
-                // callback requires output captured from 7z 
-                if (captureOutput)
+                AZStd::vector<char> fileBuffer;
+                AZStd::string fullPath = (basePath / filePathLine).c_str();
```

Comment:
This should be updated to use .Native()
```suggestion
                AZStd::string fullPath = (basePath / filePathLine).Native();
```